### PR TITLE
Determine non-purged status correctly

### PR DESCRIPTION
**Pull request by devjatkin**
Getting "installed_files" in some cases incorrectly resolve status for non-purged packages. 

Because variable "has_files" using only for determine non-purged status of package we can get status from low-level object. If status equal to apt_pkg.CURSTATE_CONFIG_FILES then package non-purged.

### CODE REVIEW

**Comment by devjatkin** [comment]
This patch fix https://github.com/ansible/ansible/issues/9582

**Comment by gregdek** [comment]
@mgwilliams Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please give it a +1; at that point, we will add the “priority_reviewed” flag to this PR, which will move it to the top of our queue for potential inclusion.

File: packaging/os/apt.py [hunk 1]
```diff
@@ -246,11 +246,9 @@ def package_status(m, pkgname, version, cache, state):
         else:
             return False, False, False
     try:
-        has_files = len(pkg.installed_files) > 0
-    except UnicodeDecodeError:
-        has_files = True
+        has_files = ll_pkg.current_state == apt_pkg.CURSTATE_CONFIG_FILES
```
Comment (bcoca) [inline packaging/os/apt.py]: should this not be a or? does current_State == apt_pkg.CURSTATE_CONFIG_FILES when there are installed files?

**Comment by gregdek** [comment]
@devjatkin looks like this PR needs to be revised according to @bcoca comments above, and also rebased.

Once you have revised and rebased this PR, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by devjatkin** [comment]
@gregdek PR rebased.
ready_for_review

**Comment by devjatkin** [comment]
@bcoca  IMHO pkg.installed_files not needed more. Has package installed files or not is not related to package status. For example, python-apt describe next statuses:
CURSTATE_NOT_INSTALLED
CURSTATE_UNPACKED
CURSTATE_HALF_CONFIGURED
CURSTATE_HALF_INSTALLED
CURSTATE_CONFIG_FILES - non-purged
CURSTATE_INSTALLED

Also I renamed "has_files" -> "has_cfg_files", as for me, that variable name more logically and uniquely.

**Comment by gregdek** [comment]
Thanks @devjatkin. 

@mgwilliams Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by mgwilliams** [comment]
@gregdek I am no longer actively using or developing Ansible, so I don't really have any input.

**Comment by gregdek** [comment]
@ansible This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Moving to core review per mgwilliams comment above. 

@devjatkin This module is now owned by the Core team, and it can take a while for patches to be reviewed. Thanks for your patience here.

**Comment by gregdek** [comment]
Thanks @devjatkin for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**